File: LeetCode/python3/974.py
class Solution:
    def subarraysDivByK(self, A, K):
        """
        :type A: List[int]
        :type K: int
        :rtype: int
        """
        count = {0: 1}
        prefix, ret = 0, 0
        for a in A:
            prefix = (prefix + a) % K
            if prefix < 0:
                prefix += K
            if prefix in count:
                ret += count[prefix]
            else:
                count[prefix] = 0
            count[prefix] += 1
        return ret

File: LeetCode/cpp/974.cpp
class Solution {
public:
    int subarraysDivByK(vector<int> &A, int K) {
        auto size = A.size();
        unordered_map<int, int> count;
        int ret = 0, prefix = 0;
        count[0] = 1;
        for (int a : A) {
            prefix = (prefix + a) % K;
            if (prefix < 0) prefix += K;
            ret += count[prefix];
            ++count[prefix];
        }
        return ret;
    }
};